src/base_classes/active_space_solver.cc

```diff
@@ -70,42 +73,58 @@ const std::map<StateInfo, std::vector<double>>& ActiveSpaceSolver::compute_energ
             method_, state, nroot, scf_info_, mo_space_info_, as_ints_, options_);
         state_method_map_[state] = method;
 
+        int twice_ms = state.twice_ms();
+        if (twice_ms < 0 and ms_avg_) {
+            psi::outfile->Printf(
+                "\n  Continue to next symmetry block: No need to compute for ms = %d / 2 < 0.",
+                twice_ms);
+            continue;
+        }
+
         method->compute_energy();
 
         const auto& energies = method->energies();
         state_energies_map_[state] = energies;
+
+        if (twice_ms > 0 and ms_avg_) {
+            StateInfo state_spin(state.nb(), state.na(), state.multiplicity(), -twice_ms,
+                                 state.irrep(), state.irrep_label());
+            state_energies_map_[state_spin] = energies;
+        }
     }
     print_energies(state_energies_map_);
     return state_energies_map_;
 }
 
 void ActiveSpaceSolver::print_energies(std::map<StateInfo, std::vector<double>>& energies) {
     print_h2("Energy Summary");
-    psi::outfile->Printf("\n    Multi.  Irrep.  No.               Energy");
-    std::string dash(41, '-');
+    psi::outfile->Printf("\n    Multi.(ms)  Irrep.  No.               Energy");
```

Comment:
Is this M_S? If yes we can just replace 
```suggestion
    psi::outfile->Printf("\n   M_S    Irrep.  No.               Energy");
```

Reply:
No, there are two values here. One is for multiplicity (i.e., Multi.) and the other is for MS that is in the parentheses.